### PR TITLE
Address Jeff feedback in llb

### PR DESCRIPTION
**Pull request by louisruch**

### CODE REVIEW

**Comment by louisruch** [comment]
Manually tested and verified that the cli bats tests run with the updates:
```
 ✓ boundary/login: can login as default user
 ✓ boundary/credential-stores: can create test static store in default project
 ✓ boundary/credential-stores: can not create already created test static store
 ✓ boundary/credential-stores: can read test static store
 ✓ boundary/credential-stores: can delete test static store
 ✓ boundary/credential-stores: can not read deleted test static store
 ✓ boundary/login: can login as default user
 ✓ boundary/credential-stores: can create credentials-test-store static store in default project
 ✓ boundary/credentials: can create test credential in credentials-test-store store
 ✓ boundary/credentials: can not create already created test credential
 ✓ boundary/credentials: can read test credential
 ✓ boundary/credentials: can delete test credential
 ✓ boundary/credential-stores: can not read deleted test credential
```

**Comment by louisruch** [comment]
@jefferai PTAL
Please note this does not address the current Vault user_password mappings only the new static credentials

**Comment by louisruch** [comment]
Sorry I should have split this into two commits instead of all in one as the `user` to `username` change touched a lot. So to be clear this PR resolves the following:

- updates `userpassword` to `usernamepassword`, similarly `user-password` to `username-password` within the new static credential workflow
- Removes references to `errors.NewDeprecate` and `errors.WrapDeprecate` for `errors.New` and `errros.Wrap` respectively 
- Removes the PasswordHmac SDK options

File: internal/credential/static/repository.go [hunk 1]
```diff
@@ -21,15 +23,15 @@ type Repository struct {
 // only be used for one transaction and it is not safe for concurrent go
 // routines to access it. WithLimit option is used as a repo wide default
 // limit applied to all ListX methods.
-func NewRepository(r db.Reader, w db.Writer, kms *kms.Kms, opt ...Option) (*Repository, error) {
+func NewRepository(ctx context.Context, r db.Reader, w db.Writer, kms *kms.Kms, opt ...Option) (*Repository, error) {
 	const op = "static.NewRepository"
 	switch {
 	case r == nil:
-		return nil, errors.NewDeprecated(errors.InvalidParameter, op, "db.Reader")
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "db.Reader")
```
Comment (jefferai) [inline internal/credential/static/repository.go:30]: Thank you!!!

**Review by jefferai — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**